tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c

```diff
@@ -1650,7 +1651,8 @@ static void test_scanx_main(void)
 
 	if ((sync_report_len != sizeof(per_adv_data1)) ||
 	    memcmp(sync_report_data, per_adv_data1, sizeof(per_adv_data1))) {
-		FAIL("Incorrect Periodic Advertising Report data.");
+		FAIL("Incorrect Periodic Advertising Report data (%u != %u).",
+		     sync_report_len, sizeof(per_adv_data1));
```

Comment:
It can also fail on the `memcmp`, and as such the `%u != %u` may not be the correct thing to log.

Reply:
Applies multiple times below as well I see.

Reply:
`memcpy` returns copy of destination pointer, hence here it is a copy of `sync_report_data`. The `sync_report_data` is a `static` global variable so the function should not fail... unless you copy faaaaaaaaaaaaar to many bytes.

BTW. This is a first place where the `sync_report_data` variable is used. The definition of the variable is ~400 lines above. Wouldn't it be better to split the test into small chunks verifying the whole thing part by part? 
It would be like reading of table of contents instead of searching for something page-by-page.

Reply:
@ppryga I am not sure I understand your comment w.r.t. `memcpy` as my comment was for `memcmp`, although it has been quite a while since I wrote that :D